src/Storage/StorageInterface.php

```diff
@@ -10,21 +10,39 @@
 interface StorageInterface
 {
     /**
-     * Persists counter value in the storage, uniquely referenced by a key with an optional expiration TTL time.
+     * Saves the value of key only if it was not already saved in the store.
+     * It returns `false` if a value was saved. Otherwise, it returns `true`
+     * and saves the value.
+     * If the storage supports expiring keys, the key will expire after the provided TTL.
      *
      * @param string $key The ID of the counter to store.
-     * @param mixed $value The value of the counter.
+     * @param float $value The value of the counter.
      * @param int $ttl The TTL value of this counter.
+     * @return bool
      */
-    public function save(string $key, mixed $value, int $ttl): void;
+    public function saveIfNotExists(string $key, float $value, int $ttl): bool;
+
+    /**
+     * Compares the old value of key to the value which was saved in the store.
+     * If it matches, method saves it to the new value and returns `true`.
+     * Otherwise, it returns `false`.
+     * If the key does not exist in the storage, it returns `false`.
+     * If the storage supports expiring keys, the key will expire after the provided TTL.
+     *
+     * @param string $key The ID of the counter to store.
+     * @param float $oldValue The old value of the counter.
+     * @param float $newValue The new value of the counter.
+     * @param int $ttl The TTL value of this counter.
+     * @return bool
+     */
+    public function saveCompareAndSwap(string $key, float $oldValue, float $newValue, int $ttl): bool;
 
     /**
      * Fetches a counter value from the storage.
      *
      * @param string $key The unique key of this counter in the storage.
-     * @param mixed $default Default value to return if the counter does not exist.
      *
-     * @return mixed The value of the counter from the storage, or $default in case of no counter with such key present.
+     * @return ?float It returns `float` if a value was saved. Otherwise, it returns `null`
      */
-    public function get(string $key, mixed $default = null): mixed;
+    public function get(string $key): ?float;
```

Comment:
Why did you delete the second parameter?

Reply:
Determine whether to use `saveIfNotExists()` or `saveCompareAndSwap()` based on the `null` return value of this method.
Default value is not used.